### DISCUSSION TITLE
How can deepMD-Kit-trained potential function model run molecular dynamics in CP2K

### DISCUSSION THREAD

**Discussion by xiangchongchen** [deepmodeling/deepmd-kit#938] [Q&A]
Dear developer：I would like to ask how to install deepMD-Kit and CP2K interface，for the trained potential function was  MD simulated in CP2K

**Comment by amcadmus** [comment]
@robinzyb may help

**Comment by robinzyb** [comment]
we recently made a pr about  cp2k/deepmd interface to cp2k repository.
https://github.com/cp2k/cp2k/pull/1620
Still need some checks from main developer group.

**Reply by Cloudac7** [reply]
> Dear developer：I would like to ask how to install deepMD-Kit and CP2K interface，for the trained potential function was MD simulated in CP2K

For now, you may try [this fork from CP2K](https://github.com/Cloudac7/cp2k.git) with `deepmd_test` branch. To install the interface, please use `install_cp2k_toochain.sh` and follow the instruction below:

- To enable the DeePMD-kit feature, `--with-deepmd` as well as `--with-tfcc` flag in toolchain should not be set to `no`. 
- Please set `--deepmd-mode` to `gpu` if GPU support is needed, and load CUDA environment before installing.
- From exist DeePMD-kit C++ interface, with `$deepmd_root` and `$tensorflow_root` (Described [here](https://docs.deepmodeling.org/projects/deepmd/en/latest/install.html#install-the-deepmd-kit-s-c-interface) in detail), by setting `--with-deepmd=$deepmd_root` and `--with-tfcc=$tensorflow_root`.
- From source code and library from Github and Anaconda Cloud, by setting the both flags to `install`. For now, DeePMD-kit v2.0.0.b1 and Tensorflow C++ interface 2.3.0 will be installed, and only **CUDA Toolkit 10.1** is OK from this way.
- From `system`, by adding the header files and library to where ld command could read directly.

**Reply by xiangchongchen** [reply]
Thank you very much for your reply. I have completed the installation of CP2K interface according to your policy. I would also like to ask how the potential function *.pb trained by DeepMD-Kit runs through CP2K. Which parameters need to be set in the cp2k.inp file.

**Reply by Cloudac7** [reply]
> Thank you very much for your reply. I have completed the installation of CP2K interface according to your policy. I would also like to ask how the potential function *.pb trained by DeepMD-Kit runs through CP2K. Which parameters need to be set in the cp2k.inp file.

Example given below, with potential file named as `graph.pb`:

```
&FORCE_EVAL
  METHOD FIST
  &MM
    &FORCEFIELD
      &CHARGE
        ATOM Au
        CHARGE 0.0
      &END
# Please set charge to 0.0 for metal.
      &NONBONDED
        &DEEPMD
          ATOMS Au Au
          POT_FILE_NAME graph.pb
          ATOM_DEEPMD_TYPE 0
        &END DEEPMD
# Element symbol should be written twice for `ATOMS`.
        &DEEPMD
          ATOMS O O
          POT_FILE_NAME graph.pb
          ATOM_DEEPMD_TYPE 1
        &END DEEPMD
      &END NONBONDED
      IGNORE_MISSING_CRITICAL_PARAMS .TRUE.
    &END FORCEFIELD
    &POISSON
      &EWALD
        EWALD_TYPE none
      &END EWALD
    &END POISSON
  &END MM
  ...
&END FORCE_EVAL
```

Just set other parts  like normal CP2K MD runs.

**Comment by xiangchongchen** [comment]
I'm glad to hear back from you, I just saw the message today due to a slight problem with my QQ email. Following your instructions I set --with-deepmd=$deepmd_root and --with-tfcc=$tensorflow_root and successfully installed deepmd and tensorflow, but I got an error when compiling cp2k:
" -D__COMPILE_REVISION="\"<unknown revision>\"" -D__DATA_DIR="\"/home/xcc/cp2k-8.2/data\"" -D__SHORT_FILE__="\"common/cp_para_types.F\"" -I'/home/xcc/cp2k-8.2/src/common/' -I'/home/xcc/cp2k-8.2/obj/local/psmp/exts/dbcsr' cp_para_types.F90
echo <unknown revision> > "/home/xcc/cp2k-8.2/obj/git-ref.tmp"
/bin/sh: -c: line 0: syntax error near unexpected token `>'
/bin/sh: -c: line 0: `echo <unknown revision> > "/home/xcc/cp2k-8.2/obj/git-ref.tmp"'
make[3]: *** [/home/xcc/cp2k-8.2/Makefile:495: /home/xcc/cp2k-8.2/obj/git-ref] Error 1
make[3]: *** Waiting for unfinished jobs....


" -D__COMPILE_REVISION="\"<unknown revision>\"" -D__DATA_DIR="\"/home/xcc/cp2k-8.2/data\"" -D__SHORT_FILE__="\"xc/xc_input_constants.F\"" -I'/home/xcc/cp2k-8.2/src/xc/' -I'/home/xcc/cp2k-8.2/obj/local/psmp/exts/dbcsr' xc_input_constants.F90
make[2]: *** [/home/xcc/cp2k-8.2/Makefile:155: all] Error 2
make[1]: *** [/home/xcc/cp2k-8.2/Makefile:137: psmp] Error 2
make: *** [Makefile:132: all] Error 2

do you know What is the cause of this?

**Reply by Cloudac7** [reply]
There might be something need to confirm. Is command `git` available from your cluster and did you add `--recursive` when cloning the repository?

**Reply by xiangchongchen** [reply]
thanks，II'm trying again
